packages/gatsby-plugin-google-analytics/src/index.js

```diff
@@ -53,4 +53,31 @@ OutboundLink.propTypes = {
   onClick: PropTypes.func,
 }
 
-export { OutboundLink }
+/**
+ * This allows the user to create custom events within their Gatsby projects.
+ * @param {string} category Required - The object that was interacted with (e.g.video)
+ * @param {string} action Required - Type of interaction (e.g. 'play')
+ * @param {string} label Optional - Useful for categorizing events (e.g. 'Spring Campaign')
+ * @param {string} value Optional - Numeric value associated with the event. (e.g. A product ID)
+ *
+ * @link https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#events
+ */
+function CustomTrackingEvent({ category, action, label, value }) {
+  if (window.ga) {
```

Comment:
Small nit: this is potentially unsafe due to `window` not existing on the server. May want to replace with something like the following: 

```suggestion
  if (typeof window !== "undefined" && window.ga) {
```

Reply:
Makes sense. I copied that check from the main function. Should get that updated as well!

Reply:
> If we're going to wrap `window.ga` we should consider incorporating the remaining `event` options that Google Analytics supports like setting `transport: beacon` and `nonInteraction: true` as well: https://developers.google.com/analytics/devguides/collection/analyticsjs/events

What do you mean by this?